src/Command/BaseCommand.php

```diff
@@ -14,15 +14,41 @@
 namespace Sonata\BlockBundle\Command;
 
 use Sonata\BlockBundle\Block\BlockServiceManagerInterface;
-use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Command\Command;
 
-abstract class BaseCommand extends ContainerAwareCommand
+abstract class BaseCommand extends Command
 {
+    /**
+     * @var BlockServiceManagerInterface
+     */
+    protected $blockManager;
+
+    public function __construct(string $name = null, BlockServiceManagerInterface $blockManager = null)
+    {
+        // NEXT_MAJOR: Remove the default value for argument 2 and the following condition
+        if (null === $blockManager) {
+            throw new \InvalidArgumentException(sprintf('Argument 2 passed to %s::%s() must be an instance of %s, %s given.', static::class, __FUNCTION__, BlockServiceManagerInterface::class, \gettype($blockManager)));
+        }
+
+        $this->blockManager = $blockManager;
+
+        parent::__construct($name);
+    }
+
     /**
      * @return BlockServiceManagerInterface
      */
     public function getBlockServiceManager()
     {
-        return $this->getContainer()->get('sonata.block.manager');
+        // NEXT_MAJOR: Remove this method
+        @trigger_error(sprintf(
+            'Method %1$s::%2$s() is deprecated since 3.x and will be removed with the 4.0 release.'.
```

Comment:
When you deprecate something, please include the package name :pray: 

```suggestion
            'Method %1$s::%2$s() is deprecated since sonata-project/block-bundle 3.x and will be removed with the 4.0 release.'.
```

Reply:
If we use method here, the deprecation message will refer to the abstract class instead of the called class. Is that fine?

Reply:
Just for context, this is similar to what is used [here](https://github.com/sonata-project/SonataAdminBundle/blob/64b83e290566dc609a7a30d6a419c1359430dfc9/src/Admin/AbstractAdmin.php#L917) to show the instantiated class name instead the abstracted one.

Reply:
Oh right, nvm then, keep it that way please.